src/stack.py

```diff
@@ -0,0 +1,44 @@
+"""Stack implementation using previously written methods- push and pop."""
+
+
+class Node(object):
+    """Where all data is stores. Points to next Node."""
+
+    def __init__(self, data=None, next=None):
+        """Class constructor to instantiate node with value and next params."""
+        self.value = data
+        self.next = next
+
+
+class Stack():
+    """Stack that pushes and pops from using existing functions."""
+
+    def __init__(self, iterable=None):
+        """Instantiate class with iterable as opitional param."""
+        self.head = None
+        self.size = 0
+        if iterable is not None:
+            for i in iterable:
+                self.push(i)
+
+    def push(self, val):
+        """Push a new node as the head of the linked list."""
+        if self.head is None:
```

Comment:
You're thinking about the right situation here, but, you actually don't even need as much code as you're using. Will there really be a difference whether `self.head` is None or an actual Node instance?